docs/articles/clustering.md

```diff
@@ -1,356 +1,3 @@
-        stats timeout 30s
-        user haproxy
-        group haproxy
-        daemon
-
-        # Default SSL material locations
-        ca-base /etc/ssl/certs
-        crt-base /etc/ssl/private
-
-        # See: https://ssl-config.mozilla.org/#server=haproxy&server-version=2.0.3&config=intermediate
-        ssl-default-bind-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384
-        ssl-default-bind-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
-        ssl-default-bind-options ssl-min-ver TLSv1.2 no-tls-tickets
-
-defaults
-        log     global
-        mode    http
-        option  httplog
-        option  dontlognull
-        timeout connect 5000
-        timeout client  50000
-        timeout server  50000
-        errorfile 400 /etc/haproxy/errors/400.http
-        errorfile 403 /etc/haproxy/errors/403.http
-        errorfile 408 /etc/haproxy/errors/408.http
-        errorfile 500 /etc/haproxy/errors/500.http
-        errorfile 502 /etc/haproxy/errors/502.http
-        errorfile 503 /etc/haproxy/errors/503.http
-        errorfile 504 /etc/haproxy/errors/504.http
-
-frontend zot
-    bind *:8080
-    mode http
-    use_backend zot-instance1 if { path_beg /v2/repo1/ }
-    use_backend zot-instance2 if { path_beg /v2/repo2/ }
-    use_backend zot-instance3 if { path_beg /v2/repo3/ }
-    default_backend zot-cluster
-
-backend zot-cluster
-    mode http
-    balance roundrobin
-    server zot-server1 127.0.0.1:9000 check
-    server zot-server2 127.0.0.2:9000 check
-    server zot-server3 127.0.0.3:9000 check
-
-backend zot-instance1
-    server zot-server1 127.0.0.1:9000 check maxconn 30
-
-backend zot-instance2
-    server zot-server2 127.0.0.2:9000 check maxconn 30
-
-backend zot-instance3
-    server zot-server3 127.0.0.3:9000 check maxconn 30
-```
-
-</details>
-
-### zot S3 configuration with DynamoDB
-
-<details>
-  <summary markdown="span">Click here to view a sample zot configuration for S3 and DynamoDB.</summary>
-
-```json
-
-{
-    "distSpecVersion": "1.0.1-dev",
-    "storage": {
-        "rootDirectory": "/tmp/zot",
-        "dedupe": false,
-        "remoteCache": true,
-        "storageDriver": {
-            "name": "s3",
-            "rootdirectory": "/zot",
-            "region": "us-east-2",
-            "bucket": "zot-storage",
-            "secure": true,
-            "skipverify": false
-        },
-        "cacheDriver": {
-            "name": "dynamodb",
-            "endpoint": "http://localhost:4566",
-            "region": "us-east-2",
-            "tableName": "MainTable"
-        }
-    },
-    "http": {
-        "address": "127.0.0.1",
-        "port": "8080"
-    },
-    "log": {
-        "level": "debug"
-    }
-}
-
-```
-</details>
-
-### zot S3 configuration with Redis
```

Comment:
I feel that the options for caching options (Redis and DynamoDB) should be documented separately from the cluster stuff because they are applicable in a non-cluster setup as well.

This applies for the session storage as well.

What do you think?

Reply:
The "storage" part is documented here https://github.com/project-zot/project-zot.github.io/blob/main/docs/articles/storage.md#cache-drivers.

I don't remember having the session configuration part documented anywhere.

Reply:
Thanks! I've documented the session configuration in this article. I think that should be alright. It is also documented in the examples Markdown in the zot repo.